lib/telephony/configuration.rb

```diff
@@ -15,6 +15,13 @@ def add_sms_config
       raise 'missing sms configuration block' unless block_given?
       sms = PinpointSmsConfiguration.new(region: 'us-west-2')
       yield sms
+
+      shortcode_country_codes = Set.new(sms.country_code_shortcodes&.keys&.map(&:to_s) || [])
+      longcode_country_codes = Set.new(sms.country_code_longcode_pool&.keys&.map(&:to_s) || [])
```

Comment:
Couple questions:

- Why `to_s`? Are we expecting mixed strings and symbols?
- Why `Set`, if a hash key can't be duplicated? (Although I suppose this could become relevant if the `to_s` could normalize to the same value)

Reply:
Although even if the above were true, maybe we'd want some regression coverage. Otherwise this would pass?

```suggestion
      shortcode_country_codes = sms.country_code_shortcodes&.keys || []
      longcode_country_codes = sms.country_code_longcode_pool&.keys || []
```

Reply:
it's very unlikely that we need to worry about keys not being a string, so I can drop that part. I thought `Set` would be useful for `#intersect?`, but it turns out that `Array` has had `#intersection` for a while.